### ISSUE TITLE
moving between std::string and OpenMS::String does not move

### ISSUE THREAD

**Issue by cbielow** [OpenMS/OpenMS#5391]
I just stumbled across this by accident, because the move in https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/FORMAT/FASTAFile.cpp#L201 is bad for performance (or so one would assume, but that's another story).
However this move never takes place. It copies (which is good for performance, but that's not what was intended here).
Why is there a copy (and a move)? Because the LHS is a `String` and the RHS is a `std::string`.

Here is a small example:
```
  String s(1000, 's');
  std::string r(500, 'r');
  
  // prints '1000 1007 vs. 500 511'
  std::cout << s.size() << " " << s.capacity() << " vs. " << r.size() << " " << r.capacity() << "\n";
  s = std::move(r);   
  // prints '500  511 vs. 500 511' --- but we expected '500 511 vs. 1000 1007'!!!
  std::cout << s.size() << " " << s.capacity() << " vs. " << r.size() << " " << r.capacity() << "\n";
```
What happened?
Well, the assignment actually called `String::String(const string& s)` with an r-value (i.e. it takes the r-value by const-ref to create a temporary `OpenMS::String`) and then moves from the temporary `String` (which has nothing to do with the std::move is the code -- you could omit that and the same thing would happen.
 
Something similar happens when you reverse the roles (assign to std::string from String):
```
  r = std::move(s); // calls basic_string& operator=(const _StringViewIsh& _Right) 
```

Ok, to solve the first problem it should be sufficient to implement:
`String operator=(std::string&&)`  (if that is even valid??)

and for the second problem its a bit more tricky but an overloaded conversion operator should do the trick
```
class String {
 ...
  operator std::string&& ()&& { printf("moving!\n");  return std::move(static_cast<std::string>(*this)); }
  operator const std::string& () const { ... }
}
```
Importantly, we need to test this actually works on all compilers (it should though).

Bottom line: never inherit from std::string! (or at least to it properly!)

**Comment by jpfeuffer** [comment]
we got rid of OpenMS::String #9450
